### PR TITLE
Increase replication factor of Cruise Control topics in none rack-aware way if there are not enough racks in the cluster.

### PR DESCRIPTION
**Pull request by kidkun**
fix for [506](https://github.com/linkedin/cruise-control/issues/506)

### CODE REVIEW

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/KafkaSampleStore.java [hunk 1]
```diff
@@ -260,9 +265,12 @@ private void maybeIncreaseTopicReplicationFactor(ZkUtils zkUtils,
       brokersByRack.get(rack).add(bm.id());
       rackByBroker.put(bm.id(), rack);
     }
+
+    boolean skipRackAwarenessCheck = false;
```
Comment (efeg) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/KafkaSampleStore.java]: Would it be possible to make `skipSampleStoreTopicRackAwarenessCheck` a configuration? We can keep the default as `true` (and throw an exception), but if the user intentionally would like to skip this check, s/he can set this config to skip this check.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/KafkaSampleStore.java [hunk 2]
```diff
@@ -260,9 +265,12 @@ private void maybeIncreaseTopicReplicationFactor(ZkUtils zkUtils,
...
     if (replicationFactor > brokersByRack.size()) {
-      throw new RuntimeException("Unable to increase topic " + topic + " replica factor to " + replicationFactor
-                                 + " since there are only " + brokersByRack.size() + " racks in the cluster.");
+      skipRackAwarenessCheck = true;
+      LOG.warn("Target replica factor for topic " + topic + " is larger than number of racks in cluster, new replica maybe" +
```
Comment (efeg) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/KafkaSampleStore.java]: `replica ` -> `replication`

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/KafkaSampleStore.java (hunk 1 above)
Comment (kidkun) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/KafkaSampleStore.java]: I believe you mean default value to be `false`.

Comment (efeg) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/KafkaSampleStore.java]: Correct, I meant to say `false`.

**Pull request merged**